web/src/message_fetch.js

```diff
@@ -428,9 +429,33 @@ export function maybe_load_older_messages(opts) {
     if (!msg_list_data.fetch_status.can_load_older_messages()) {
         // We may already be loading old messages or already
         // got the oldest one.
+        if (opts.recent_view) {
+            recent_view_ui.set_backfill_in_progress(false);
+        }
         return;
     }
 
+    if (opts.recent_view) {
```

Comment:
This block should have an `&& is_backfill_in_progress` check, right? Otherwise, the button will be a noop once we reach this point.

Reply:
``` diff
diff --git a/web/src/message_fetch.js b/web/src/message_fetch.js
index 64367320a4..57d0bad4f7 100644
--- a/web/src/message_fetch.js
+++ b/web/src/message_fetch.js
@@ -435,7 +435,15 @@ export function maybe_load_older_messages(opts) {
         return;
     }
 
-    if (opts.recent_view) {
+    if (opts.recent_view && recent_view_ui.is_backfill_in_progress) {
+        // The recent view "load more" button does a tail-recursive
+        // backfill, so that one doesn't have to click dozens of times
+        // to get a large amount of history.
+        //
+        // We can afford to do this in our server load budget, because
+        // this is a rare operation; prior to May 2024, Zulip fetched
+        // all history since the oldest unread unconditionally.
+
         let fetched_substantial_history = false;
         if (msg_list_data.num_items() >= consts.recent_view_minimum_load_more_fetch_size) {
             fetched_substantial_history = true;
```